added test application for nativenet transceiver

OlegHahm opened the pull request:
This project can be used with desvirt framework.

OlegHahm committed:
added test application for nativenet transceiver

LudwigKnuepfer commented:
I think defining sender and receiver targets in the Makefile would make sense.. if I got right how this is intended to be used.

OlegHahm commented:
It was intended to use make with command line parameters, but you're right: dedicated targets in the Makefile would be better. I'll add this.

mehlis commented on test_nativenet/main.c:
```diff
@@ -0,0 +1,142 @@
+
+    p.length = PACKET_SIZE;
+    p.dst = 0;
+    snd_buffer[0] = SENDER_ADDR;
```
why not `p.src` ?

OlegHahm replied:
Unfortunately nativenet does not yet havea source address field in the packet header.

LudwigKnuepfer replied:
Make an issue and I'll implement it ;-)

mehlis replied:
did https://github.com/RIOT-OS/projects/issues/29

mehlis commented on test_nativenet/main.c:
```diff
@@ -0,0 +1,142 @@
+    puts("Start sending packets");
+    while (1) {
+        snd_buffer[1] = (i & 0xFF00) >> 8;
+        snd_buffer[2] = i & 0x00FF;
```
please define two macros for the right hand code

mehlis replied:
what does this do? you are using byte 1 and 2 for some magic...the left hand needs clarification

OlegHahm replied:
Agreed.
(The code is just about sending the 16 bit sequence number as payload.)

mehlis commented:
this test project depends on #29 to be solved.

don't merge now.

OlegHahm committed:
added sender target to Makefile

OlegHahm committed:
make use of the nativenet source address (as introduced in #170)

OlegHahm commented:
According to your comments I've added the corresponding Makefile target (receiver is the default target) and changed to use the nativenet source address field in the packet header.

OlegHahm commented:
Further comments on this?

mehlis commented:
ok for me

LudwigKnuepfer commented:
go
